framework/sandstone.cpp

```diff
@@ -1038,6 +1038,18 @@ int num_packages() {
     return Topology::topology().packages.size();
 }
 
+void reschedule() {
+    // Let's choose a random CPU to hop to
+    int cpu_idx = random32() % num_cpus();
```

Comment:
I think we ought to make this a queue or implement different strategies (as a test config option/knob). The reason for the queue is: it helps avoiding oversubscription of a HW for a prolonged period of time which will help the test execution time and may reveal different problems. I think we should have both, actually.